### PR TITLE
Fix zmount test cases

### PR DESCRIPTION
**Pull request by samaradel**
### Description

Fix zmount test suite expected errors.

### Changes
image

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3523

### Tested Scenarios

- Update the value of size to get errors

### Checklist

- [x] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 1]
```diff
@@ -15,29 +15,43 @@ describe("Zmount module", () => {
 
     zmount.size = size;
 
-    const result = () => zmount.challenge();
+    const result = zmount.challenge();
 
-    expect(result).toThrow();
+    expect(result).toBe(size.toString());
```
Comment (xmonader) [inline packages/grid_client/tests/modules/zmount.test.ts]: can directly do expect(zmount.challenge()).toBe....

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 2]
```diff
@@ -15,29 +15,43 @@ describe("Zmount module", () => {
...
   });
 
   test("Max value for size.", () => {
     const size = 100 * 1025 ** 4;
```
Comment (xmonader) [inline packages/grid_client/tests/modules/zmount.test.ts]: What do these numbers mean? try extracting to a const

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 3]
```diff
@@ -15,29 +15,43 @@ describe("Zmount module", () => {
...
+      expect.objectContaining({
+        constraints: {
+          max: "size must not be greater than 10995116277760",
+        },
+      }),
+    );
   });
 
   test("Size doesn't accept decimal value.", () => {
     const size = 1.5;
```
Comment (xmonader) [inline packages/grid_client/tests/modules/zmount.test.ts]: is that 1.5 byte?

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 4]
```diff
@@ -15,29 +15,43 @@ describe("Zmount module", () => {
...
+      expect.objectContaining({
+        constraints: {
+          isInt: "size must be an integer number",
+          min: "size must not be less than 104857600",
+        },
+      }),
+    );
   });
 
   test("Size empty value.", () => {
```
Comment (xmonader) [inline packages/grid_client/tests/modules/zmount.test.ts]: While we are at it, check against null, undefined, NaN, {} , and []

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 5]
```diff
@@ -15,29 +15,43 @@ describe("Zmount module", () => {
...
     const result = () => zmount.challenge();
 
-    expect(result).toThrow();
+    expect(() => {
+      zmount.size = size;
+      result;
+    }).toThrow(
+      expect.objectContaining({
+        constraints: {
+          max: "size must not be greater than 10995116277760",
```
Comment (xmonader) [inline packages/grid_client/tests/modules/zmount.test.ts]: check against the boundaries max, and min

and off by 1 min - 1 and max+1

**Review by xmonader — CHANGES_REQUESTED** [review]
We need to use better descriptions to describe the cases better

**Review by 0oM4R — APPROVED** [review]
passed locally, not sure about the tests description 
image

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 6]
```diff
@@ -1,58 +1,39 @@
...
+  beforeEach(() => {
+    zmount = new Zmount();
   });
 
-  test("Size doesn't accept decimal value.", () => {
-    const size = 1.5;
-
-    zmount.size = size;
+  describe("size property validation", () => {
+    it("should fail validation if size is less than the minimum", async () => {
```
Comment (amiraabouhadid) [inline packages/grid_client/tests/modules/zmount.test.ts:36]: shouldn't we have a test to test if input size is int first? before testing min and max

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 7]
```diff
@@ -1,58 +1,39 @@
...
-    zmount.size = negative_size;
-
-    const result = () => zmount.challenge();
+  describe("challenge method", () => {
+    it("should return the size as a string", () => {
+      zmount.size = 5 * 1024 ** 3;
 
-    expect(result).toThrow();
+      const result = zmount.challenge();
+      expect(result).toBe("5368709120");
```
Comment (amiraabouhadid) [inline packages/grid_client/tests/modules/zmount.test.ts]: without checking if size is int , the challenge can return any special chars in the string, no?

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 8]
```diff
@@ -1,58 +1,39 @@
...
+  describe("size property validation", () => {
+    it("should fail validation if size is less than the minimum", async () => {
+      // Less than 100 MB
+      expect(() => (zmount.size = 50 * 1024 ** 2)).toThrow();
+    });
 
-    const result = () => zmount.challenge();
+    it("should fail validation if size is greater than the maximum", async () => {
+      // Greater than 10 TB
+      expect(() => (zmount.size = 15 * 1024 ** 4)).toThrow();
```
Comment (amiraabouhadid) [inline packages/grid_client/tests/modules/zmount.test.ts]: define the min and max as constants before any of the tests

**Review by ramezsaeed — APPROVED** [review]

**Comment by amiraabouhadid** [comment]
please place the min and max values in a constant before any of the tests. Please move tests that check if size is null, Nan or not an int before the min and max tests

**Review by maayarosama — APPROVED** [review]

**Review by amiraabouhadid — APPROVED** [review]

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 9]
```diff
@@ -1,58 +1,58 @@
+import { validate } from "class-validator";
+
 import { Zmount } from "../../src";
 
-let zmount: Zmount;
+describe("Zmount Class", () => {
+  let zmount: Zmount;
+  const minSize = 50 * 1024 ** 2; // Less than 100 MB
+  const maxSize = 15 * 1024 ** 4; // Greater than 10 TB
+  const validSize = 5 * 1024 ** 3;
```
Comment (AhmedHanafy725) [inline packages/grid_client/tests/modules/zmount.test.ts]: it's better to have a random size (within a certain range) to test with

you can use this method to generate the random number https://github.com/threefoldtech/tfgrid-sdk-ts/blob/f9a5b06796073bcd61ef729240308d122beb3646/packages/grid_client/src/helpers/utils.ts#L18

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
you should add a test for setting a floating number

**Review by ramezsaeed — APPROVED** [review]

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 10]
```diff
@@ -1,58 +1,60 @@
...
-    zmount.size = size;
+import { Zmount } from "../../src";
+import { getRandomNumber } from "../../src/helpers/utils";
 
-    const result = () => zmount.challenge();
+describe("Zmount Class", () => {
+  let zmount: Zmount;
+  const minSize = 50 * 1024 ** 2; // Less than 100 MB
+  const maxSize = 15 * 1024 ** 4; // Greater than 10 TB
+  const validSize = 5 * 1024 ** 3;
```
Comment (maayarosama) [inline packages/grid_client/tests/modules/zmount.test.ts]: Why are we initializing this with a value, if we're using getRandomNumber

File: packages/grid_client/tests/modules/zmount.test.ts [hunk 11]
```diff
@@ -1,58 +1,59 @@
...
-    zmount.size = negative_size;
-
-    const result = () => zmount.challenge();
+  describe("challenge method", () => {
+    it("should return the size as a string", () => {
+      zmount.size = validSize;
 
-    expect(result).toThrow();
+      const result = zmount.challenge();
+      expect(typeof result).toBe("string");
```
Comment (AhmedHanafy725) [inline packages/grid_client/tests/modules/zmount.test.ts]: you can cast `validSize` to string and check against its value

**Review by AhmedHanafy725 — APPROVED** [review]

**Review by maayarosama — APPROVED** [review]

**Pull request merged**